docs/developer-guide/working-with-plugins.md

```diff
@@ -84,6 +84,23 @@ array corresponds to the part that was returned from the `preprocess` method. Th
 You can have both rules and processors in a single plugin. You can also have multiple processors in one plugin.
 To support multiple extensions, add each one to the `processors` element and point them to the same object.
 
+### Configs in Plugins
+
+You can bundle configurations inside a plugin. This can be useful when you want to provide not just code style, but also some custom rules to support it. You can specify configurations under `configs` key. Please note that when exposing configurations, you have to name each one, and there is no default. So your users will have to specify the name of the configuration they want to use.
+
+```js
+configs: {
+    default: {
```

Comment:
Might be better to either change this example config name (since there's no default config), or allow default configs named `default`.